pkg/storage/replica_command.go

```diff
@@ -2062,6 +2062,10 @@ func evalRequestLease(
 			return newFailedLeaseTrigger(false /* isTransfer */), rErr
 		}
 		if args.Lease.Type() == roachpb.LeaseExpiration {
+			// NB: Avoid mutating pointers in the argument which might be shared with
+			// the caller.
+			t := *args.Lease.Expiration
```

Comment:
This still smells, we shouldn't be changing `args` at all. Just make `args := *(previous stuff)` so that we really have a local copy here (in addition to your change here). But we also need to switch out `Lease` if it's a pointer field.